12-watchtower-API.md

```diff
@@ -127,122 +129,167 @@ This message contains all the information regarding the appointment that the cli
+* **Encrypted transaction**: The `cipher`,`encrypted_blob` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
+* **Customer signature** The `customer_public_key` and `customer_signature` provides an explicit message about the job from the customer.  The `customer_public_key` can also be used for refunds if applicable.
+
 The transaction `locator` can be deterministically computed by both the client and the server. Locators of wrong size are therefore invalid.
 
 `start_block` and `end_block` too close to the current chain tip may result in the tower missing the trigger and therefore should be avoided.
 
-Too far away is a subjective concept. Towers accepting jobs that will start in the far future or that may last a really long time risk having to store data for long periods of time.
+Too far away is a subjective concept. Towers accepting jobs that start in the far future or that may last a really long time risk having to store data for long periods of time and should, therefore, by avoided.
 
-The concept of too small for `dispute_delta` is also subjective. The `dispute_delta` defines how many blocks the server will have to respond with the `justice_transaction`. The smallest the value, the more the server risks to fail the appointment.
+The concept of too small for `dispute_delta` is also subjective. The `dispute_delta` defines how many blocks the server will have to respond with the `justice_transaction` after a breach is seen. The smallest the value, the more the server risks to fail the appointment.
 
-`encrypted_blob` should have been encrypted using `cipher`. Block ciphers have a size multiple of the block length, which depends on the key size. Therefore some incorrect `encrypted_blob` can be spotted checking the `transaction_size`. Moreover, `encrypted_blob` have to be at least as big as:
+The `encrypted_blob` should have been encrypted using `cipher`. Block ciphers have a size multiple of the block length, which depends on the key size. Therefore some incorrect `encrypted_blob` can be spotted checking the `transaction_size`. Moreover, `encrypted_blob` have to be at least as big as:
 
-`ceil(minimum_viable_transaction_size / cipher_block_size)`
+`cipher_block_size * ceil(minimum_viable_transaction_size / cipher_block_size)`
 
 And at most as big as:
 
-`ceil(maximum_viable_transaction_size / cipher_block_size`) 
+`cipher_block_size * ceil(maximum_viable_transaction_size / cipher_block_size`) 
 
-`minimum_viable_transaction_size` and `maximum_viable_transaction_size` refer to the minimum/maximum size required to create a valid transaction.
+`minimum_viable_transaction_size` and `maximum_viable_transaction_size` refer to the minimum/maximum size required to create a valid transaction. Accepting `encrypted_blob` outside those boundaries will ease DoS attacks on the server.
```

Comment:
makes me wonder if we need an error message to say "encrypted blog too big". so we place a limit on the size of a transaction / blob. some of them with lots of HTLC can easily grow.

Reply:
No error messages have been defined so far (there are a few), so not for this iteration. It may be interesting for future ones.

Reply:
we will need error messages though for the standard?

Reply:
i.e. client needs to know why it was rejected and deal with it somehow?

Reply:
I think it would be interesting to have error messages, yeah. An automatic thing I can foresee for example is that the fee for a job is too low, so the wallet could renegotiate the fee with the other side of the channel and resend the appointment request

Reply:
You mean leak, right? I already discussed that with @yahgwai back in the day, and ended up reaching the conclusion that high expressivity in this cases may be good to avoid poor UX. I guess we can propose it and then discuss with the community.

I'll write a non-exhaustive error section to have something to discuss in the first draft